pkg/virtctl/create/vm/vm.go

```diff
@@ -499,11 +543,78 @@ func (c *createVM) addDiskWithBootOrder(flag string, vm *v1.VirtualMachine, name
 	return nil
 }
 
+func (c *createVM) inferFromVolume(vm *v1.VirtualMachine) error {
+	if c.inferInstancetype && c.instancetype == "" {
+		if err := c.withInferredInstancetype(vm); err != nil && c.explicitInstancetypeInference {
```

Comment:
This is a little confusing, I get that we only want to surface the error if the inferFromVolume flags are used but we shouldn't be running withInferredInstancetype unless required right?

Reply:
`inferFromVolume` is always called last after applying all other options to the VM because we necessarily don't know if we should be running with `withInferredInstancetype` or not. If either `c.inferInstancetype` (with `--infer-instancetype=false`) or `c.instancetype` (with `--instancetype=my-instancetype`) were present, we don't want to infer the instancetype, but otherwise we want to do it implicitly by default. So in other words, we are only running with `withInferredInstancetype` when required, but we always run with `inferFromVolume`.